gpytorch/kernels/matern_kernel.py

```diff
@@ -25,11 +26,31 @@ def __init__(
             ard_num_dims=ard_num_dims,
             log_lengthscale_prior=log_lengthscale_prior,
             active_dims=active_dims,
+            batch_size=batch_size,
             log_lengthscale_bounds=log_lengthscale_bounds,
         )
         self.nu = nu
         self.eps = eps
 
+    def forward_diag(self, x1, x2):
```

Comment:
Could we add a unit test for this method?

Reply:
Going to open a separate issue for this because we need to make unit tests for matern kernel in general, and this PR fixes #246